cloud/azure/azure_rm_virtualmachine.py

```diff
@@ -0,0 +1,1316 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2016 Matt Davis, <mdavis@ansible.com>
+#                    Chris Houseknecht, <house@redhat.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+DOCUMENTATION = '''
+---
+module: azure_rm_virtualmachine
+
+version_added: "2.1"
+
+short_description: Manage Azure virtual machines.
+
+description:
+    - Create, update, stop and start a virtual machine. Provide an existing storage account and network interface or
+      allow the module to create these for you. If you choose not to provide a network interface, the resource group
+      must contain a virtual network with at least one subnet.
+    - Currently requires an image found in the Azure Marketplace. Use azure_rm_virtualmachineimage_facts module
+      to discover the publisher, offer, sku and version of a particular image.
+
+options:
+    resource_group:
+        description:
+            - Name of the resource group containing the virtual machine.
+        required: true
+    name:
+        description:
+            - Name of the virtual machine.
+        required: true
+    state:
+        description:
+            - Assert the state of the virtual machine.
+            - State 'present' will check that the machine exists with the requested configuration. If the configuration
+              of the existing machine does not match, the machine will be updated. Use options start, stop,
+              deallocate and restart to change the machine's power state.
+            - State 'absent' will remove the virtual machine.
+        default: present
+        required: false
+        choices:
+            - absent
+            - present
+    start:
```

Comment:
i did not mean to create 'action options' stareted=yes|no is a more proper way of expressing this, always think in 'declarative mode'